service/controller/v21/templates/cloudformation/guest/vpc.go

```diff
@@ -23,4 +23,26 @@ const VPC = `{{define "vpc"}}
       Tags:
         - Key: Name
           Value: {{ $v.ClusterID }}
+  VPCS3Endpoint:
+    Type: 'AWS::EC2::VPCEndpoint'
+    Properties:
+      VpcId: !Ref VPC
+      RouteTableIds:
+        {{- range $v.RouteTableNames }}
+        - !Ref {{ .ResourceName }}
+        {{- end}}
+      ServiceName: 'com.amazonaws.{{ $v.Region }}.s3'
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: "{{ $v.ClusterID }}-vpc-s3-endpoint-policy-bucket"
+            Principal : "*"
+            Effect: "Allow"
+            Action: "*"
+            Resource: "arn:{{ $v.RegionARN }}:s3::{{ $v.GuestAccountID }}:*"
+          - Sid: "{{ $v.ClusterID }}-vpc-s3-endpoint-policy-object"
```

Comment:
The current policy seems not correct because according the aws arn guideline it should not contain the `GuestAccountID` as showed here https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html

A good policy could be this:

```yaml
PolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Sid: "{{ $v.ClusterID }}-vpc-s3-endpoint-policy-bucket"
            Principal: "*"
            Effect: "Allow"
            Action: "s3:*"
            Resource:
              - "arn:{{ $v.RegionARN }}:s3:::*"
              - "arn:{{ $v.RegionARN }}:s3:::*/*"
```

Reply:
this is a problem

Reply:
Actually thinking for 2 more secs this Condition doesn't cover all the cases :-(